### DISCUSSION TITLE
Request the command which could check specific text

### DISCUSSION THREAD

**Discussion by shukuo** [ipxe/ipxe#326] [Ideas]
I would like to use smbios to determine specific string to chain different file. So create this discussion to request the iPXE command which could check the string contain specific text.

For example:
If **"iscon"** is this new request command and **${smbios/product}** is **"abc 123"**

iPXE> iscon ${smbios/product} abc && chain 123.ipxe || chain 456.ipxe
The result will chain 123.ipxe file

iPXE> iscon ${smbios/product} 666 && chain 123.ipxe || chain 456.ipxe
The result will chain 456.ipxe file

**Comment by mcb30** [comment]
The iPXE scripting language is not intended to be a general-purpose programming language, and the number of features that it supports is deliberately kept small in order to reduce code size.  For the kind of processing that you are trying to do, you will need to implement the logic on the server side.  For example:

```
chain http://server/select.cgi?product=${smbios/product}
```

**Reply by shukuo** [reply]
Got it. By the way, could you share more about the cgi script example for iPXE?
I am trying to use cgi method but it will got error message "Could not boot: Exec format error (http://ipxe.org/2e008081)"
I tried to open it directly with browser it could work. Does there is any specific item need to setting?

Boot.ipxe:

> #!ipxe
> chain select.cgi

select.cgi:

> #!/usr/bin/perl
> use strict;
> use warnings;
> print "Content-type: text/html\n\n";
> print "hello";

**Reply by NiKiZe** [reply]
If you try to chain to that URL, what is returned must be a valid iPXE script (or other valid data that can be executed) just as with any other chain that is done from iPXE.

**Reply by shukuo** [reply]
Does cgi file is the valid data for iPXE?

**Reply by shukuo** [reply]
I already figure out how to use cgi file with iPXE.
